jobs/payment-jobs/tasks/direct_pay_automated_refund_task.py

```diff
@@ -135,21 +136,30 @@ def _refund_complete(cls, invoice: Invoice):
     @staticmethod
     def _is_glstatus_rejected(status: OrderStatus) -> bool:
         """Check for bad refundglstatus."""
-        return any(line.refundglstatus == PaymentDetailsGlStatus.RJCT
-                   for line in status.revenue)
+        return any(refund_data.refundglstatus == PaymentDetailsGlStatus.RJCT
+                   for line in status.revenue for refund_data in line.refund_data)
 
     @staticmethod
     def _is_status_paid_and_invoice_refund_requested(status: OrderStatus, invoice: Invoice) -> bool:
         """Check for successful refund and invoice status = REFUND_REQUESTED."""
-        return all(line.refundglstatus == PaymentDetailsGlStatus.PAID
-                   for line in status.revenue) \
-            and invoice.invoice_status_code == InvoiceStatus.REFUND_REQUESTED.value
+        for line in status.revenue:
+            if len(line.refund_data) == 0:
```

Comment:
Had to put this in, otherwise using `all` like above fails when `refund_data = []`